### ISSUE TITLE
macOS wheels don't work on macOS 10.12

### ISSUE THREAD

**Issue by RazerM** [libgit2/pygit2#1026]
The wheels report 10.9+ compatibility but don't work on 10.12. The same error happens for the 1.2.0 and 1.2.1 wheels:

```
Python 3.7.1 (default, Apr  7 2020, 18:00:22)
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygit2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...lib/python3.7/site-packages/pygit2/__init__.py", line 30, in <module>
    from ._pygit2 import *
ImportError: dlopen(...lib/python3.7/site-packages/pygit2/_pygit2.cpython-37m-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: ...lib/python3.7/site-packages/pygit2/.dylibs/libgit2.1.0.0.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in ...lib/python3.7/site-packages/pygit2/.dylibs/libgit2.1.0.0.dylib
```

I guess they use a symbol from a newer libSystem.

(I know the workaround is to do `pip install --no-binary pygit2 pygit2`).

In case it's related, see https://github.com/pyca/cryptography/issues/5215 and https://github.com/pyca/infra/pull/275

**Comment by jdavid** [comment]
ping @rcoup

**Comment by rcoup** [comment]
@RazerM I'm a little out of my depth here, but can you have a quick try of the wheels produced from [2833dc1](https://github.com/rcoup/pygit2/actions/runs/197670928)? 

And can you run some diagnostics under macos10.9 for me?

```console
$ otool -L /path/to/pygit2/_pygit2.cpython-37m-darwin.so
$ otool -l /path/to/pygit2/_pygit2.cpython-37m-darwin.so | grep -A4 VER
$ otool -L /path/to/pygit2/.dylibs/libgit2.1.0.0.dylib
$ otool -l /path/to/pygit2/.dylibs/libgit2.1.0.0.dylib | grep -A4 VER
```

**Comment by RazerM** [comment]
The wheel you linked to works!

I assume you want the diagnostics from the pygit2 I compiled from source (?):

```
$ otool -L _pygit2.cpython-37m-darwin.so
_pygit2.cpython-37m-darwin.so:
  /usr/local/opt/libgit2/lib/libgit2.1.0.dylib (compatibility version 1.0.0, current version 1.0.1)
  /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)

$ otool -l _pygit2.cpython-37m-darwin.so | grep -A4 VER
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.12
      sdk 10.12
Load command 8
--
      cmd LC_SOURCE_VERSION
  cmdsize 16
  version 0.0
Load command 9
          cmd LC_LOAD_DYLIB

$ otool -L .dylibs/libgit2.1.0.1.dylib
.dylibs/libgit2.1.0.1.dylib:
	/DLC/pygit2/libgit2.1.0.1.dylib (compatibility version 1.0.0, current version 1.0.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1677.104.0)
	/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 59306.140.5)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)

$ otool -l .dylibs/libgit2.1.0.1.dylib | grep -A4 VER
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.9
      sdk 10.15.6
Load command 9
--
      cmd LC_SOURCE_VERSION
  cmdsize 16
  version 0.0
Load command 10
          cmd LC_LOAD_DYLIB
```

**Comment by rcoup** [comment]
> The wheel you linked to works!

That seems suspiciously easy... 🕵️ 

There are some warnings during the [libgit2 compilation](https://github.com/rcoup/pygit2/actions/runs/197670928):

```
/Users/runner/work/pygit2/pygit2/libgit2/src/unix/posix.h:86:9: warning: 'futimens' is only available on macOS 10.13 or newer [-Wunguarded-availability-new]
        return futimens(f, s);
               ^~~~~~~~
/Applications/Xcode_11.6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/stat.h:396:9: note: 'futimens' has been marked as being introduced in macOS 10.13 here, but the deployment target is macOS 10.9.0
int     futimens(int __fd, const struct timespec __times[2]) __API_AVAILABLE(macosx(10.13), ios(11.0), tvos(11.0), watchos(4.0));
        ^
/Users/runner/work/pygit2/pygit2/libgit2/src/unix/posix.h:86:9: note: enclose 'futimens' in a __builtin_available check to silence this warning
        return futimens(f, s);
               ^~~~~~~~
1 warning generated.
```

Seems like `futimens()` is only [used in win32 though](https://github.com/libgit2/libgit2/search?q=p_futimes&type=Code), so might be safe enough to ignore.

Libgit2 cmake actually checks for it explicitly:
```
-- Looking for futimens
-- Looking for futimens - found
```
though https://gitlab.kitware.com/cmake/cmake/-/issues/17101#note_297983 suggests the CheckFunctionExists test in cmake doesn't actually respect macOS deployment targets.

@jdavid happy to commit this change anyway?

**Comment by jdavid** [comment]
> @jdavid happy to commit this change anyway?

@rcoup yes, will you do a PR?
